test/performance/config/performance-reporter.ts

```diff
@@ -72,26 +92,46 @@ export function curateResults(
 	results: WPRawPerformanceResults
 ): WPPerformanceResults {
 	const output = {
-		timeToFirstByte: median( results.timeToFirstByte ),
```

Comment:
This will change the reporting from `median` to `average`, is that intentional?

Reply:
> In some cases we also seems to be calculating the median, for others the average

It seems to me like a pointless inconsistency. The metrics measured don't look fundamentally different from each other. That why I'm trying to unify the stats to use only one methods.

Are you worried that changing it would invalidate the stats history? I think in our case the medians and averages should be approx the same, any difference is just a random noise. I'll check if one of them is systematically higher than the other one.

Reply:
If we remove that, maybe we should also remove the `median` function etc.

Reply:
> this would be reverting #54157

That's true, we'll have to deal with that. The reason why median is needed to "stabilize the metric" is that there are outliers and that median is less sensitive to them than average.

This is a real-world example of an `inserterHover` measurement with 20 data points:
```
[2.60, 3.09, 2.57, 2.62, 2.21, 2.98, 2.34, 3.34, 2.37, 3.00, 2.14, 2.16, 2.07, 2.22, 2.12, 2.27, 2.66, 2.20, 6.89, 2.99]
```
Almost all measurements are nicely grouped around 2.5, in a ±0.5 range, but there is one outlier at 6.8:

<img width="576" alt="inserter-data" src="https://github.com/WordPress/gutenberg/assets/664258/655da293-917c-4333-81d1-43c3b52682f0">

The presence of this outlier moves the median from 2.37 to 2.47, by 0.1. But the average is more sensitive, it's moved from 2.52 to 2.74, by 0.22.

When there are more outliers than one, then the stats are even more distorted.

We can solve this by detecting and removing the outliers. By arguing that when a certain result is too far from all others, it means that the measurement instrument broke down and produced a garbage result. Without outliers, neither the median and the average are distorted, and are closer to each other.

My ultimate goal here is to do some significance test for each metric and produce a result like:
```
on trunk       the inserterHover metric is 2.5±0.5ms
on this branch the inserterHover metric is 1.5±0.5ms
with 95% confidence this is a real improvement, not a random noise
```
This enables automated reporting in PR where the tool can confidently say "out of 100 metrics I checked, your PR improves this one but another two are significantly worse"

All the common significance form accept averages and variances as inputs. Not medians and quartiles. That's why I'm trying to move to averages.